codex/sales.nim

```diff
@@ -193,9 +190,12 @@ proc processSlot(
 
   agent.onCleanUp = proc(
       reprocessSlot = false, returnedCollateral = UInt256.none
-  ) {.async.} =
+  ) {.async: (raises: []).} =
     trace "slot cleanup"
-    await sales.cleanUp(agent, reprocessSlot, returnedCollateral)
+    try:
+      await sales.cleanUp(agent, reprocessSlot, returnedCollateral)
+    except CancelledError as e:
+      trace "slot cleanup was cancelled", error = e.msgDetail
```

Comment:
This won't be needed once `cleanUp` no longer raises. We don't want `cleanUp` to be cancelled in any circumstance, no matter who calls it.